homeassistant/components/dobiss/light.py

```diff
@@ -0,0 +1,169 @@
+            async_dispatcher_connect(self.hass, DOMAIN, self.signal_handler)
+        )
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Entity being removed from hass."""
+        self._dobisslight.remove_callback(self.async_write_ha_state)
+
+    async def signal_handler(self, data: dict[str, Any]) -> None:
+        """Handle domain-specific signal by calling appropriate method."""
+        entity_ids = data[ATTR_ENTITY_ID]
+
+        if entity_ids == ENTITY_MATCH_NONE:
+            return
+
+        if entity_ids == ENTITY_MATCH_ALL or self.entity_id in entity_ids:
+            params = {
+                key: value
+                for key, value in data.items()
+                if key not in ["entity_id", "method"]
+            }
+            await getattr(self, data["method"])(**params)
+
+    async def turn_on_service(
+        self,
+        brightness: int | None = None,
+        delayon: int | None = None,
+        delayoff: int | None = None,
+        from_pir: bool = False,
+    ) -> None:
+        """Turn on the light with optional parameters."""
+        await self._dobisslight.turn_on(
+            brightness=brightness, delayon=delayon, delayoff=delayoff, from_pir=from_pir
+        )
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness of the light."""
+        if not self._dobisslight.dimmable:
+            return None
+        # dobiss works from 0-100, ha from 0-255
+        return int((self._dobisslight.value / 100) * 255)
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if light is on."""
+        return self._dobisslight.is_on
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on or control the light."""
+        # dobiss works from 0-100, ha from 0-255
+        raw_brightness = kwargs.get(ATTR_BRIGHTNESS)
+        if raw_brightness is not None:
+            brightness = int((raw_brightness / 255) * 100)
+        else:
+            brightness = 100
+        await self._dobisslight.turn_on(brightness)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Instruct the light to turn off."""
+        await self._dobisslight.turn_off()
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use in the frontend."""
+        if isinstance(self._dobisslight, DobissAnalogOutput):
+            return "mdi:hvac"
+        return super().icon or "mdi:lightbulb"
+
+    @property
+    def color_mode(self) -> ColorMode:
+        """Return the color mode of the light."""
+        if self._dobisslight.dimmable:
+            return ColorMode.BRIGHTNESS
+        return ColorMode.ONOFF
+
+    @property
+    def supported_color_modes(self) -> set[ColorMode]:
+        """Return the supported color modes."""
+        return self._attr_supported_color_modes
+
+    @property
+    def name(self) -> str:
+        """Return the display name of this light."""
+        return self._dobisslight.name
```

Comment:
new integrations are required to have `_attr_has_entity_name` set to True, and in this case you want this variable to be the device name, and then set the entity name to `None`